pkg/machinery/resources/cluster/endpoints.go

```diff
@@ -0,0 +1,75 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package cluster
+
+import (
+	"fmt"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/protobuf"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+
+	"github.com/siderolabs/talos/pkg/machinery/proto"
+)
+
+// EndpointsType is type of Endpoints resource.
```

Comment:
I feel we should move this resource to `k8s/`, as it's really specific there (with the loadbalancer itself later).

And I would it call it some more specific way, e.g. `APIServerEndpoints`, as we have many other endpoints as well.